csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+            do not have a specific validator.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+
+    """
+
+    type_: Literal[
+        TypeEnum.OBJECT,
+        TypeEnum.ARRAY,
+        TypeEnum.YEAR,
+        TypeEnum.YEARMONTH,
+        TypeEnum.DURATION,
+        TypeEnum.ANY,
+        None,
+    ] = Field(None, alias="type", description="A string specifying the type.")
+    format_: FormatDefault | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class StringColumnValidator(CommonValidator):
+    """Validator for the string columns in the Table Schema.
+
+    This class is used to validate the string columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'string'.
+        format_: A string specifying a format, with valid values being 'default',
+            'email', 'uri', 'binary', 'uuid' and None.
+
+    """
+
+    type_: Literal[TypeEnum.STRING] = Field(
+        TypeEnum.STRING, alias="type", description="A string specifying the type."
+    )
+    format_: FormatString | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class NumberColumnValidator(CommonValidator):
+    """Validator for the number columns in the Table Schema.
+
+    This class is used to validate the number columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'number'.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+        decimalChar: The character used to separate the integer and fractional. If
+            None, '.' is used.
+        groupChar: The character used to separate groups of thousands. None assumed.
+        bareNumber: If True, the number is a bare number, without any formatting. If
+            None, True is used. If false the contents of this field may contain leading
+            and/or trailing non-numeric characters that must be stripped to parse the
+            number, eg. '95%', '£50.5'.
+
+    """
+
+    type_: Literal[TypeEnum.NUMBER] = Field(
+        TypeEnum.NUMBER, alias="type", description="A string specifying the type."
+    )
+    format_: FormatDefault | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+    decimalChar: str | None = Field(
+        None,
+        description="The character used to separate the integer and fractional. "
+        + "If None, '.' is used.",
+    )
+    groupChar: str | None = Field(
+        None, description="The character used to separate groups of thousands."
+    )
+    bareNumber: bool | None = Field(
```

Comment:
Why the camelCase here? Are these the names used in the spec?

Reply:
Exactly. Probably I should make these aliases and use standard snake case here, to be honest, but I just thought it would be best to keep the name of the descriptors as close as possible to the specification to avoid confusion. Not sure. What do you think?
```python
bare_number = Field(None, alias="bareNumber", description="...")
```

Reply:
Open an issue about this.